### PR TITLE
Add karma-dev setup

### PR DESCRIPTION
**Pull request by alandawi**
- Turns off coverage preprocessors and reporting
- Enables watch mode
- Add mocha style test reporting
- Update Package.json with new dependencies

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -75,18 +77,25 @@
     "archiver": "^1.0.0",
     "awesome-typescript-loader": "2.0.1",
     "chai": "^3.5.0",
+    "chai-as-promised": "^5.3.0",
+    "core-js": "^2.4.1",
     "coveralls": "^2.11.6",
     "es6-promise": "^3.2.1",
     "es6-promisify": "^4.1.0",
     "es6-promisify-all": "^0.1.0",
+    "history": "^3.0.0",
```
Comment (mediamonks-arjan) [inline package.json]: This doesn't belong in this update (but was in the seng-router commit)

File: package.json [hunk 2]
```diff
@@ -75,18 +77,25 @@
     "archiver": "^1.0.0",
     "awesome-typescript-loader": "2.0.1",
     "chai": "^3.5.0",
+    "chai-as-promised": "^5.3.0",
+    "core-js": "^2.4.1",
```
Comment (mediamonks-arjan) [inline package.json]: This doesn't belong in this update (but was in the seng-router commit)

**Comment by mediamonks-arjan** [comment]
I think you're also missing some typings that were in the initial commit in seng-router:
https://github.com/mediamonks/seng-router/blob/master/typings.json

File: typings.json [hunk 3]
```diff
@@ -3,9 +3,13 @@
     "mocha": "registry:dt/mocha#2.2.5+20160619032855"
   },
   "devDependencies": {
-    "chai": "registry:npm/chai#3.5.0+20160415060238"
+    "chai": "registry:npm/chai#3.5.0+20160415060238",
+    "es6-promise": "registry:npm/es6-promise#3.0.0+20160723033700",
```
Comment (ThaNarie) [inline typings.json]: This one should not be in the boilerplate.

File: typings.json [hunk 4]
```diff
@@ -3,9 +3,13 @@
...
+    "sinon": "registry:npm/sinon#1.16.0+20160723033700",
+    "sinon-chai": "registry:npm/sinon-chai#2.8.0+20160310030142"
   },
   "globalDependencies": {
-    "webpack-env": "registry:dt/webpack-env#1.12.2+20160316155526"
+    "webpack-env": "registry:dt/webpack-env#1.12.2+20160316155526",
+    "xregexp": "registry:dt/xregexp#3.0.0+20160317120654"
```
Comment (ThaNarie) [inline typings.json]: This one should not be in the boilerplate.

Comment (alandawi) [inline typings.json]: I revert the commit.

Comment (mediamonks-arjan) [inline typings.json]: `sinon` and `sinon-chai` should be in there, so don't revert everything :)

**Comment by ThaNarie** [comment]
I've tested the setup locally, and there is just 1 mismatch left, `chai-as-promised` is referenced in `karma.conf.dev.js`, but not in the `package.json` (and is also missing in `typings.json`).

I know a lot of code comments from me were about removing things, but maybe we should just include the full test-setup in the boilerplate so anyone can start using sinon spies and chai promises without extra setup costs.
(history and  es6-promise should be left out)

Adding them would then include:
- sinon
- sinon-chai
- chai-as-promised
- mocha

**Comment by alandawi** [comment]
@ThaNarie @flut1 For me it´s ok. If you don´t want to use the full-test, you can remove it. But if you need it, no need to add it.

**Comment by art0rz** [comment]
I disagree. We should not include these libraries by default, even if they are commonly used. The boilerplate should be as bare-bones as possible. Perhaps an entry in the wiki outlining integration of these modules would be better.

**Comment by alandawi** [comment]
@art0rz make really a difference if don´t include those libraries? Anyway, the wiki is not a bad idea.

**Comment by art0rz** [comment]
It does not make a big difference practically, but I see no reason to include a library when you're not using it.

**Comment by ThaNarie** [comment]
Okay, lets round up this PR :)

Inlcude by default (karma config, npm, typings):
- mocha
- chai

Exclude, but document on wiki:
- chai-as-promised
- sinon
- sinon-chai

@alandawi Since this commit history is a little messy (adding and reverting), please reset (mixed mode) your master branch to the point before you started (but leave all the changed files in your working copy), do the required changes, and a single commit covering all the changes. Then force-push that branch again so this PR will be clean. :)

**Pull request closed without merging**